### PR TITLE
ForceAsyncAwaiter Tests

### PR DESCRIPTION
**Pull request by mikaelm12**
Some tests for the ForceAsyncAwaiter

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs [hunk 1]
```diff
@@ -160,6 +160,22 @@ public class HubConnectionTests
...
 
+        [Fact]
+        public async Task CheckSyncContext()
+        {
+            SynchronizationContext.SetSynchronizationContext(new SynchronizationContext());
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new LoggerFactory());
+            Assert.NotNull(SynchronizationContext.Current);
+            await hubConnection.StartAsync();
+            Assert.Null(SynchronizationContext.Current);
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: This looks wrong. That doesn't happen...

Comment (mikaelm12) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: Yeah, tbh I was kind of confused by this behavior

Comment (mikaelm12) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: How should we be testing this at the HubConnection level?

**Comment by mikaelm12** [comment]
There were issues with using the default SyncContext. Things should look better now.

**Comment by moozzyk** [comment]
I think it would be better if - instead of/in addition to using event handlers - you created a mock transport (possibly wrapping an existing transport if it makes it easier) and checking sync context in the transport methods. If you use this pattern you can also test HubConnection async methods by mocking IConnection.

**Comment by mikaelm12** [comment]
I was able to add tests at the HubConnection level by checking the SyncContexts in the Connected and Closed events. Do you think I should still mock the transport and implement the tests that way? I'd have to brush up on how to use Mocks 😅

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs [hunk 2]
```diff
@@ -160,6 +160,50 @@ public class HubConnectionTests
...
+        [Fact]
+        public async Task CheckSyncContextOnClosed()
+        {
+            var hubConnection = new HubConnection(new TestConnection());
+            var closedEventTcs = new TaskCompletionSource<Exception>();
+            hubConnection.Closed += async e =>
+            {
+                await Task.Yield();
+                Assert.Null(SynchronizationContext.Current);
+                await Task.CompletedTask;
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: You don't need to await Task.CompletedTask

**Comment by muratg** [comment]
@mikaelm12 *ping*

**Comment by mikaelm12** [comment]
Closing for now as this isn't critical atm

**Pull request closed without merging**